app/Http/Controllers/BookController.php

```diff
@@ -12,24 +14,49 @@ public function __construct()
         $this->bookService = app(BookServices::class);
     }
 
-    public function store(Request $request) {
+    public function store(Request $request)
+    {
         $upload = $this->bookService->store($request);
-        if($upload !== true) {
+
+        if ($upload !== true) {
             return response()->json([
                 'status' => 422,
                 'message' => $upload,
                 'data' => []
             ]);
-        }
-        else {
-            return response()->json([
+        } else return response()->json([
                 'status' => 200,
                 'message' => 'SUCCESS',
                 'data' => []
             ]);
-           
-        }  
-     
-      
+        
+    }
+    public function getBookList(GetBookListRequest $request)
+    {
+        $params = $request->all();
+        $pageNum = $params['page'];
+        $itemsPerPage = null;
+
+        if ($request->has('items_per_page')) {
+            $itemsPerPage = $params['items_per_page'];
+        }
+
+        $filterCol = $request->has('filter_col');
+        $filterValue = $request->has('filter_value');
+
+        if ($filterCol && $filterValue) {
+            $filterCol = $params['filter_col'];
+
+            $filterValue = $params['filter_value'];
+        } else $filterCol = null;
```

Comment:
```suggestion
        } else {
         $filterCol = null;
         }
```

thêm ngoặc nhọn vào e, js thì dùng kiểu này được còn laravel là nó báo không hợp lệ đó